packages/faucet/src/faucet.ts

```diff
@@ -35,38 +43,28 @@ export class Faucet {
 
   private readonly tokenConfig: TokenConfiguration;
   private readonly tokenManager: TokenManager;
-  private readonly readOnlyClient: CosmosClient;
-  private readonly clients: { [senderAddress: string]: SigningCosmosClient };
+  private readonly readOnlyClient: CosmosClient | StargateClient;
+  private readonly clients: { [senderAddress: string]: SigningCosmosClient | SigningStargateClient };
   private readonly logging: boolean;
   private creditCount = 0;
 
   private constructor(
-    apiUrl: string,
     addressPrefix: string,
     config: TokenConfiguration,
     wallets: ReadonlyArray<readonly [string, OfflineSigner]>,
+    clients: { [senderAddress: string]: SigningCosmosClient | SigningStargateClient },
```

Comment:
If we pass in the clients here, we don't need `wallets` anymore. The addresses are available in the client's `senderAddress`. We just need to preserve the order of the clients.